numba/parfors/array_analysis.py

```diff
@@ -1325,6 +1325,20 @@ def _analyze_inst(self, label, scope, equiv_set, inst, redefined):
                     shape = gvalue
                 elif isinstance(gvalue, int):
                     shape = (gvalue,)
+            elif isinstance(inst.value, ir.Arg):
+                if (
+                    isinstance(typ, types.containers.UniTuple)
+                    and isinstance(typ.dtype, types.Integer)
+                ):
+                    shape = inst.value
+                elif (
```

Comment:
This branch doesn't appear to be taken by the new unittests?

Reply:
I added some tests that are supposed to take these paths and I confirmed it in isolated tests locally.